src/_pytest/fixtures.py

```diff
@@ -862,19 +862,17 @@ def addfinalizer(self, finalizer):
         self._finalizers.append(finalizer)
 
     def finish(self, request):
-        exceptions = []
+        exc = None
         try:
             while self._finalizers:
                 try:
                     func = self._finalizers.pop()
                     func()
-                except:  # noqa
-                    exceptions.append(sys.exc_info())
-            if exceptions:
-                _, val, tb = exceptions[0]
-                # Ensure to not keep frame references through traceback.
-                del exceptions
-                raise val.with_traceback(tb)
+                except BaseException as e:
+                    if exc is None:
+                        exc = e
+            if exc:
+                raise exc
```

Comment:
Looks like the initial idea here was to collect them for reporting them all somehow?  (see also https://github.com/pytest-dev/pytest/pull/6868/files#diff-4d6d922c3b97a9eae157f8f653627b13R320-R321 - maybe add a TODO/XXX comment like there here also?)
Anyway, simplifying it like this is nice.